### PR TITLE
add patch as http method

### PR DESCRIPTION
**Pull request by Rrahmey**
- Added patch method as part of HTTP request methods

### CODE REVIEW

**Comment by Pearapps** [comment]
Can we have test coverage for all of this stuff?

**Comment by Rrahmey** [comment]
@Pearapps, tests were just added

File: Classes/TMRequestHelpers/TMRequestParamaterizer.m [hunk 1]
```diff
@@ -145,7 +145,7 @@ - (nonnull NSURLRequest *)signedURLRequestWithRequest:(nonnull NSURLRequest *)UR
         tokenSecret = self.userCredentials.tokenSecret;
     }
 
-    NSDictionary *postParameters = (self.request.method == TMHTTPRequestMethodPOST && [self.request.requestBody encodeParameters]) ? [self.request.requestBody parameters] : nil;
+    NSDictionary *postParameters = (self.request.method == (TMHTTPRequestMethodPOST | TMHTTPRequestMethodPATCH) && [self.request.requestBody encodeParameters]) ? [self.request.requestBody parameters] : nil;
```
Comment (Pearapps) [inline Classes/TMRequestHelpers/TMRequestParamaterizer.m]: Is this logic something we can test?

**Comment by Rrahmey** [comment]
@Pearapps tests are written for the change I made :)

**Comment by Pearapps** [comment]
Looks good - will deploy this and a new version tomorrow

**Pull request merged**